node/external/transactionAPI/gasUsedAndFeeProcessor.go

```diff
@@ -49,6 +93,101 @@ func (gfp *gasUsedAndFeeProcessor) computeAndAttachGasUsedAndFee(tx *transaction
 	gfp.prepareTxWithResultsBasedOnLogs(tx, hasRefundForSender)
 }
 
+func (gfp *gasUsedAndFeeProcessor) getFeeOfRelayed(tx *transaction.ApiTransactionResult) (*big.Int, bool) {
+	if !tx.IsRelayed {
+		return nil, false
+	}
+
+	if len(tx.InnerTransactions) > 0 {
+		return gfp.feeComputer.ComputeTransactionFee(tx), true
+	}
+
+	if len(tx.Data) == 0 {
+		return nil, false
+	}
+
+	funcName, args, err := gfp.argsParser.ParseCallData(string(tx.Data))
+	if err != nil {
+		return nil, false
+	}
+
+	if funcName == core.RelayedTransaction {
+		return gfp.handleRelayedV1(args, tx)
+	}
+
+	if funcName == core.RelayedTransactionV2 {
+		return gfp.handleRelayedV2(args, tx)
+	}
+
+	return nil, false
+}
+
+func (gfp *gasUsedAndFeeProcessor) handleRelayedV1(args [][]byte, tx *transaction.ApiTransactionResult) (*big.Int, bool) {
+	if len(args) != 1 {
+		return nil, false
+	}
+
+	innerTx := &transaction.Transaction{}
+	err := gfp.marshaller.Unmarshal(innerTx, args[0])
+	if err != nil {
+		return nil, false
+	}
+
+	gasUsed := gfp.feeComputer.ComputeGasLimit(tx)
+	fee := gfp.feeComputer.ComputeTxFeeBasedOnGasUsed(tx, gasUsed)
+
+	innerFee := gfp.feeComputer.ComputeTransactionFee(&transaction.ApiTransactionResult{
+		Tx: innerTx,
+	})
+
+	return big.NewInt(0).Add(fee, innerFee), true
+}
+
+func (gfp *gasUsedAndFeeProcessor) handleRelayedV2(args [][]byte, tx *transaction.ApiTransactionResult) (*big.Int, bool) {
```

Comment:
If I understood well, a similar logic exists in the actual processing flow, as well.

This can be considered duplicated code (and maybe it can be kept within `economics data`).

Not a request to change, just an idea.